File: app/src/main/java/io/plan8/backoffice/util/NetworkUtil.java
package io.plan8.backoffice.util;

import android.content.Context;
import android.net.ConnectivityManager;

/**
 * Created by chokwanghwan on 2017. 11. 28..
 */

public class NetworkUtil {
    private static volatile NetworkUtil instance = null;

    public static NetworkUtil getInstance() {
        if (null == instance) {
            synchronized (NetworkUtil.class) {
                instance = new NetworkUtil();
            }
        }
        return instance;
    }

    public boolean checkInternetConnection(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        boolean result = false;
        if (null != connectivityManager
                && null != connectivityManager.getActiveNetworkInfo()
                && connectivityManager.getActiveNetworkInfo().isAvailable()
                && connectivityManager.getActiveNetworkInfo().isConnected()) {
            result = true;
        }
        return result;
    }
}

File: app/src/main/java/io/plan8/business/util/NetworkUtil.kt
package io.plan8.business.util

import android.content.Context
import android.net.ConnectivityManager

/**
 * Created by SSozi on 2017. 11. 2..
 */
object NetworkUtil {
    fun checkInternetConnection(context: Context): Boolean {

        val conManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        return conManager.activeNetworkInfo != null
                && conManager.activeNetworkInfo.isAvailable
                && conManager.activeNetworkInfo.isConnected
    }
}